File: 2685-first-completely-painted-row-or-column/2685-first-completely-painted-row-or-column.py
class Solution(object):
    def firstCompleteIndex(self, arr, mat):
        """
        :type arr: List[int]
        :type mat: List[List[int]]
        :rtype: int
        """
        m, n = len(mat), len(mat[0])
        pos = {mat[i][j]: (i, j) for i in range(m) for j in range(n)}
        row, col = [0] * m, [0] * n
        for i, num in enumerate(arr):
            r, c = pos[num]
            row[r] += 1
            col[c] += 1
            if row[r] == n or col[c] == m:
                return i
        return -1

File: 2685-first-completely-painted-row-or-column/2685-first-completely-painted-row-or-column.java
class Solution {
    public int firstCompleteIndex(int[] arr, int[][] mat) {
        int m = mat.length, n = mat[0].length;
        Map<Integer, Integer> pos = new HashMap<>();
        
        for (int i = 0; i < m; i++)
            for (int j = 0; j < n; j++)
                pos.put(mat[i][j], i * n + j);

                
        int[] row = new int[m], col = new int[n];
        for (int i = 0; i < arr.length; i++) {
            int index = pos.get(arr[i]);
            int r = index / n, c = index % n;
            if (++row[r] == n || ++col[c] == m) return i;
        }

        return -1;
    }
}
